### ISSUE TITLE
UI bug with code preview

### ISSUE THREAD

**Issue by theamanjs** [vuejs/docs-next#384]
## About Bug
I visited the latest Vue 3 [documentation](https://v3.vuejs.org/guide/) and saw a UI bug in the documentation related to code highlighter. I saw this probably due to my screen resolution which is `1366 X 768`. Specifically, the row number and the code palette were not in co-ordination for reference screenshots are attached below. 

### To reproduce the bug, you can check the documentation with the specified resolution on any page. 

The screenshots are of [Composition-API Page](https://v3.vuejs.org/guide/composition-api-introduction.html). This issue was on every page but its level of inconsistency varies just like in below screenshots.

![Screenshot_2020-08-05_19-45-16](https://user-images.githubusercontent.com/38748298/89424489-772f4500-d755-11ea-8d31-886e8e9c00fe.png)

![Screenshot_2020-08-05_19-45-51](https://user-images.githubusercontent.com/38748298/89424486-75658180-d755-11ea-8c34-e3b01ba5397e.png)

**Comment by marina-mosti** [comment]
Hey @theamanjs I tried to reproduce this with the exact size but I dont see the problem. Can you provide browser details?

<img width="1488" alt="image" src="https://user-images.githubusercontent.com/14843771/89717663-b86f6100-d9b0-11ea-8746-aa651b2fe4a2.png">

**Comment by bencodezen** [comment]
@theamanjs I just checked as well and agree with @marina-mosti's findings in being unable to reproduce the issue on Chrome, Firefox or Safari.

**Comment by theamanjs** [comment]
@marina-mosti @bencodezen I'm using Chrome `Version 83.0.4103.97 (Official Build) (64-bit)`. I've attached a video showing problem. P.S. I noticed When I zoom out or zoom in it revert to normal, but on 100% zoom, it shows me this bug. 

I was not able to upload the video here, so I uploaded it on the drive. If you find it comfortable to see on Google drive it is fine. Otherwise, I will upload it somewhere else where you will find it comfortable to watch [Kazam_screencast_00016.mp4](https://drive.google.com/file/d/1rGDN8usp7l40UxMUqwuz6a834TplhBEs/view?usp=sharing)

**Comment by tommyip** [comment]
I am having the same issue with the same screen resolution of 1366x768 in Chrome. Firefox does not have this problem but some of the highlighted lines are misaligned. 

![image](https://user-images.githubusercontent.com/18504232/90013604-2c26ac00-dcd8-11ea-9e2b-50fa34028d39.png)

---

Chrome: Version 84.0.4147.85 (Official Build) beta (64-bit)
Firefox: 79.0 (64-bit)
OS: Linux Mint 20 Cinnamon

**Comment by skirtles-code** [comment]
So, the culprit appears to be this:

```stylus
code, kbd, .line-number
  font-family source-code-pro, Menlo, Monaco, Consolas, "Courier New", monospace
```

That's from `src/.vuepress/theme/styles/index.styl`.

Whether you see the problem will depend on which font your browser ends up using. I'm on Ubuntu and that uses `Courier New`, which has very obvious alignment problems in Chrome. I could reproduce the problem on Windows and OS X by forcing them to use `Courier New` too:

```stylus
code, kbd, .line-number
  font-family "Courier New"
```

That doesn't quite make the various platforms misalign in exactly the same way but it's pretty close.

From what I can tell, the main problem is that the `<br>` elements between the line numbers are using a different font and that can cause them to have the wrong height. There's a similar problem getting the highlighting to line up, though that is less extreme.

This seemed to work across all platforms for me:

```stylus
code, kbd, .line-numbers-wrapper, .highlight-lines, pre[class*=language-]
  font-family source-code-pro, Menlo, Monaco, Consolas, "Courier New", monospace
```

There didn't seem to be convenient classes on all the elements I needed, so I've ended up with something that looks a bit horrific. It makes all three of the main wrapper elements use the same font.

@bencodezen I'm not familiar enough with how everything fits together but it looks to me like this is something that should be fixed in VuePress first, rather than just patching the docs?

**Comment by bencodezen** [comment]
@skirtles-code Agreed that this should be patched at the VuePress level. Based on your findings, are you recommending that we remove Courier New as a fallback font?

**Comment by skirtles-code** [comment]
@bencodezen I don't think it's really a problem with `Courier New`, that just happens to be the font that exposes the problem.

Perhaps this picture will help to explain:

![Affected elements](https://user-images.githubusercontent.com/65301168/95702891-8a72fb00-0c45-11eb-8a96-ae7a9c078925.png)

The elements I've marked with red circles are currently using the fixed-width font. The other elements are using the variable-width font.

As there's a mixture of fonts, the heights don't necessarily line up unless the two fonts happen to have the same metrics.

The fix I proposed in my previous comment applied the fixed-width font to the 3 elements that I've marked with blue squares. The other elements inherit it. I couldn't put it on the root element because the `::before` needs to use the variable-width font.

**Comment by NataliaTepluhina** [comment]
I'm moving this issue to VuePress repo as it's not a docs issue indeed.
